index-scheduler/src/index_mapper.rs

```diff
@@ -66,15 +66,30 @@ impl IndexMapper {
 
     /// Create or open an index in the specified path.
     /// The path *must* exists or an error will be thrown.
-    fn create_or_open_index(&self, path: &Path) -> Result<Index> {
+    fn create_or_open_index(
+        &self,
+        path: &Path,
+        date: Option<(time::OffsetDateTime, time::OffsetDateTime)>,
+    ) -> Result<Index> {
         let mut options = EnvOpenOptions::new();
         options.map_size(clamp_to_page_size(self.index_size));
         options.max_readers(1024);
-        Ok(Index::new(options, path)?)
+
+        if date == None {
+            Ok(Index::new(options, path)?)
+        } else {
+            let (created, updated) = date.unwrap();
+            Ok(Index::new_with_creation_dates(options, path, created, updated)?)
+        }
```

Comment:
```suggestion
        if let Some(date) = date {
            Ok(Index::new_with_creation_dates(options, path, created, updated)?)
        } else {
            Ok(Index::new(options, path)?)
        }
```

When we can we try to avoid using `.unwrap`